st2common/st2common/models/api/execution.py

```diff
@@ -136,7 +136,10 @@ def to_model(cls, instance):
             if not value and not cls.model._fields[attr].required:
                 continue
             if attr not in ActionExecutionAPI.SKIP:
-                setattr(model, attr, value)
-        model.start_timestamp = isotime.parse(instance.start_timestamp)
-        model.end_timestamp = isotime.parse(instance.end_timestamp)
+                values[attr] = value
+
+        values['start_timestamp'] = isotime.parse(instance.start_timestamp)
+        values['end_timestamp'] = isotime.parse(instance.end_timestamp)
+
+        model = cls.model(**values)
```

Comment:
I really don't get why this is a better pattern. This clearly looks like an expensive approach to construction given you create an extra object.

Is there some benefit to this approach that I am missing?